### PR TITLE
Feature/cce 502 new tree

### PR DESCRIPTION
**Pull request by nicandra**
Changes:

* follow the new visuals and logic for desktop and mobile layout
* complete separation between old tree style and the new tree style (introduce an y-tree main class)
* add styling (and use already existing variables, colors)
* remove inline styles
* complete refactoring and simplification of the markup (remove unused classes, replace clearfix with row, etc)
* add the new tree file in the patterns list (we can discuss the order, title and what description the file should have)

Descoped:

* logic for the expand

### CODE REVIEW

**Comment by nicandra** [comment]
Added the changes from Saad's feedback:
* remove background color
* remove paddings
* add borders (the cleanest is by adding left and right border and the "divider" class to the markup where it was needed)
* break long lines so we don't see horizontal scroll; 

Addition: 
* remove the margins for the row so the component is aligned with the text in the page

File: src/less/globals/themes/_techne.less [hunk 1]
```diff
@@ -151,6 +151,17 @@
...
+
+//** Headers text color
+@tablehead-text-color: @default-greyblue;
+
+//** Rows text-color
+@tablerow-text-color: @dark-blue;
+
+//** Primary data for on focus rows text
+@primarydata-on-focus-text-color: @default-blue;
+
```
Comment (saadmhybris) [inline src/less/globals/themes/_techne.less]: can you please also add these vars to src/less/globals/themes/_belize.less.

we need to maintail a 1:1 vars with these two files as to not miss anything when we eventually update the belize theme.

File: src/less/techne-patterns/_trees.less [hunk 2]
```diff
@@ -1,3 +1,401 @@
...
+//     <div class="row tablehead hidden-sm hidden-xs">
+//         <div class="col-sm-4">
+//             <div class="col-sm-1 ">    
+//                 <a class="btn btn-link category-toggle ">
+//                     <span class="glyphicon glyphicon-chevron-down" ></span>
+//                 </a>
+//             </div>
+//             <div class="col-sm-10">Data Header</div>
+//         </div>
+//         <div class="col-sm-3 ng-binding">Data Header</div>
```
Comment (saadmhybris) [inline src/less/techne-patterns/_trees.less]: we need to remove all instanced of this class "ng-binding"

File: src/less/techne-patterns/_trees.less [hunk 3]
```diff
@@ -1,3 +1,401 @@
...
+//             <a class="btn btn-link category-toggle ">
+//                 <span class="glyphicon glyphicon-chevron-down" ></span>
+//             </a>
+//         </div>
+//         <div class="col-xs-8 ng-binding">Data Header</div>
+//     </div>
+//     <div class="divider"></div>
+//     <!-- nodes -->
+//     <!-- node 1 -->
+//     <div class="row mobileLayout visible-xs visible-sm ">
```
Comment (nicandra) [inline src/less/techne-patterns/_trees.less]: do we really want to change all the existing class names (in common with the old tree) and force everyone to do a full markup update?

File: src/less/globals/themes/_techne.less (hunk 1 above)
Comment (nicandra) [inline src/less/globals/themes/_techne.less]: done

File: src/less/techne-patterns/_trees.less (hunk 2 above)
Comment (nicandra) [inline src/less/techne-patterns/_trees.less]: done

File: src/less/techne-patterns/_trees.less [hunk 4]
```diff
@@ -1,3 +1,401 @@
+// new tree
+//
+// this is the new tree pattern
+//
+// Markup:
+// <div class="y-tree">
+//     <!-- headers -->
+//     <div class="row tablehead hidden-sm hidden-xs">
```
Comment (nicandra) [inline src/less/techne-patterns/_trees.less]: ok for tree-head, changed it

File: src/less/techne-patterns/_trees.less [hunk 5]
```diff
@@ -1,3 +1,401 @@
...
+//             <a class="btn btn-link category-toggle ">
+//                 <span class="glyphicon glyphicon-chevron-up"></span>
+//             </a>
+//         </div>
+//         <div class="col-xs-8">
+//             <div class="mobileData">
+//                 <span class="tablehead">Data Header</span>
+//                 <p class="primaryData">&ltPrimary Value&gt</p>
+//             </div>
+//             <div class="mobileData">
```
Comment (nicandra) [inline src/less/techne-patterns/_trees.less]: done; all the new classes should be name consistent now

**Comment by SAPMontreal** [comment]
I see one bug in the expand/collapse:
expand from the header expander -> will expand whole tree
collapse the first Parent Row
collapse the tree from the header expand/collapse icon -> one parent is expanding while the other is collapsing

**Comment by saadmhybris** [comment]
closing this PR without merge as there is still work to be done on this. Will open a new one once the pattern is ready.

**Pull request closed without merging**